core/src/main/java/kafka/server/SharePartition.java

```diff
@@ -769,6 +885,10 @@ private AcquiredRecords acquireNewBatchRecords(String memberId, long baseOffset,
                 RecordState.ACQUIRED,
                 1,
                 timerTask));
+            // if the cachedState is empty, i.e. startOffset is -1, then startOffset needs to be updated
+            if (startOffset == -1)  {
+                startOffset = baseOffset;
+            }
```

Comment:
I don't think we should introduce concept of -1 in startOffset to track things. It should be the one pointing to the offset till which the records in partition are acked successfully (Acknowledged or Archived).